src/services/ledger/receive.ts

```diff
@@ -178,7 +204,16 @@ const addReceiptFee = async ({
       .credit(bankOwnerPath, fee, metadata)
 
     const savedEntry = await entry.commit()
-    return translateToLedgerJournal(savedEntry)
+    const journalEntry = translateToLedgerJournal(savedEntry)
+
+    journalEntry.transactionIds.map((_id) =>
+      txMetadataRepo.persistnew({
+        id: _id,
```

Comment:
`_id` => by convention `_` represents a mongoose / MongoDB native id.
`id` without `_` represents a domain representation
and the conversion happens with `from/toObjectId<Type>`